codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostStandardRequestDecoder.java

```diff
@@ -569,7 +571,7 @@ private void parseBodyAttributes() {
                                 currentStatus = MultiPartStatus.PREEPILOGUE;
                                 ampersandpos = currentpos - 2;
                                 sao.setReadPosition(0);
-                                setFinalBuffer(undecodedChunk.retainedSlice(firstpos, ampersandpos - firstpos));
+                                setFinalBuffer(undecodedChunk.copy(firstpos, ampersandpos - firstpos));
```

Comment:
hmm... retainedSlice(...) should be fine. This will do an extra memory copy which I think is not needed.